vllm/model_executor/weight_utils.py

```diff
@@ -275,6 +275,107 @@ def hf_model_weights_iterator(
             torch.cuda.empty_cache()
 
 
+def kv_cache_scales_loader(
```

Comment:
is this the best way to distribute the scaling factors? i wonder whether there can be a convention to include it in the model's weights dictionary, similar to how other quantization methods support it.

Reply:
Most other methods is weight quantization only, we sure looked.
Here, we first introduced scaling factors to KV cache (not weight sort), soon we will also add scaling factors for activations (tensor `X`s). We found adding them as extended parameters this way making sense and also light weight.